src/test/resources/mapping/expectedTransformationFields.json

```diff
@@ -68,44 +82,58 @@
 			"fieldId": "instance.identifiers.gpo.item.number",
 			"displayNameKey": "instance.identifiers",
 			"referenceDataValue": "GPO item number",
-			"path": "$.instance[*].identifiers[?(@identifierTypeId==351ebc1c-3aae-4825-8765-c6d50dbf011f)].value",
+			"path": "$.instance[*].identifiers[?(@.identifierTypeId==351ebc1c-3aae-4825-8765-c6d50dbf011f)].value",
 			"recordType": "INSTANCE"
 		},
 		{
 			"fieldId": "instance.identifiers.isbn",
 			"displayNameKey": "instance.identifiers",
 			"referenceDataValue": "ISBN",
-			"path": "$.instance[*].identifiers[?(@identifierTypeId==8261054f-be78-422d-bd51-4ed9f33c3422)].value",
+			"path": "$.instance[*].identifiers[?(@.identifierTypeId==8261054f-be78-422d-bd51-4ed9f33c3422)].value",
 			"recordType": "INSTANCE"
 		},
 		{
 			"fieldId": "instance.identifiers.issn",
 			"displayNameKey": "instance.identifiers",
 			"referenceDataValue": "ISSN",
-			"path": "$.instance[*].identifiers[?(@identifierTypeId==913300b2-03ed-469a-8179-c1092c991227)].value",
+			"path": "$.instance[*].identifiers[?(@.identifierTypeId==913300b2-03ed-469a-8179-c1092c991227)].value",
 			"recordType": "INSTANCE"
 		},
 		{
 			"fieldId": "instance.identifiers.lccn",
 			"displayNameKey": "instance.identifiers",
 			"referenceDataValue": "LCCN",
-			"path": "$.instance[*].identifiers[?(@identifierTypeId==c858e4f2-2b6b-4385-842b-60732ee14abb)].value",
+			"path": "$.instance[*].identifiers[?(@.identifierTypeId==c858e4f2-2b6b-4385-842b-60732ee14abb)].value",
 			"recordType": "INSTANCE"
 		},
 		{
 			"fieldId": "instance.identifiers.oclc",
 			"displayNameKey": "instance.identifiers",
 			"referenceDataValue": "OCLC",
-			"path": "$.instance[*].identifiers[?(@identifierTypeId==439bfbae-75bc-4f74-9fc7-b2a2d47ce3ef)].value",
+			"path": "$.instance[*].identifiers[?(@.identifierTypeId==439bfbae-75bc-4f74-9fc7-b2a2d47ce3ef)].value",
 			"recordType": "INSTANCE"
 		},
 		{
 			"fieldId": "instance.identifiers.system.control.number",
 			"displayNameKey": "instance.identifiers",
 			"referenceDataValue": "System control number",
-			"path": "$.instance[*].identifiers[?(@identifierTypeId==7e591197-f335-4afb-bc6d-a6d76ca3bace)].value",
+			"path": "$.instance[*].identifiers[?(@.identifierTypeId==7e591197-f335-4afb-bc6d-a6d76ca3bace)].value",
 			"recordType": "INSTANCE"
 		},
+    {
+      "fieldId":"instance.instancetypeid.other",
+      "displayNameKey":"instance.instanceTypeId",
```

Comment:
This will need a translation funtion right, because the Id must be translated to value

Reply:
yes, this work required in stories about applying. We have a map in the rule factory, there we will add field id as a key, and function name as a value

Reply:
added function

Reply:
The other functions are still missing, but we can add it in the implementation